lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -3178,6 +3185,15 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
     if (!PD->getPrimaryAssociatedTypeNames().empty())
       (void) PD->getPrimaryAssociatedTypes();
 
+    // We cannot compute the requirement signature of a protocol
```

Comment:
Something else might trigger computation of the requirement signature though, even if we skip it here.

Reply:
The change you suggested to `getParentForLookup` solves this, so we don't need to skip it anymore.